main.go

```diff
@@ -428,67 +377,94 @@ func main() {
+		if err != nil {
+			failf("Failed to find APK or AAB files, error: %s", err)
+		}
+		appFiles = append(appFiles, matches...)
 	}
 
-	if len(apkFiles) == 0 {
-		log.Warnf("No file name matched apk filters")
+	if len(appFiles) == 0 {
+		log.Warnf("No file name matched app filters")
 	}
 
-	lastCopiedApkFile := ""
-	copiedApkFiles := []string{}
-	for _, apkFile := range apkFiles {
-		fi, err := os.Lstat(apkFile)
+	var copiedApkFiles []string
+	var copiedAabFiles []string
+	for _, appFile := range appFiles {
+		fi, err := os.Lstat(appFile)
 		if err != nil {
 			failf("Failed to get file info, error: %s", err)
 		}
 
 		if fi.ModTime().Before(gradleStarted) {
-			log.Warnf("skipping: %s, modified before the gradle task has started", apkFile)
+			log.Warnf("skipping: %s, modified before the gradle task has started", appFile)
 			continue
 		}
 
-		ext := filepath.Ext(apkFile)
-		baseName := filepath.Base(apkFile)
+		ext := filepath.Ext(appFile)
+		baseName := filepath.Base(appFile)
 		baseName = strings.TrimSuffix(baseName, ext)
 
 		deployPth, err := findDeployPth(configs.DeployDir, baseName, ext)
 		if err != nil {
 			failf("Failed to create apk deploy path, error: %s", err)
 		}
 
-		log.Printf("copy %s to %s", apkFile, deployPth)
-		if err := command.CopyFile(apkFile, deployPth); err != nil {
+		log.Printf("copy %s to %s", appFile, deployPth)
+		if err := command.CopyFile(appFile, deployPth); err != nil {
 			failf("Failed to copy apk, error: %s", err)
 		}
 
-		lastCopiedApkFile = deployPth
-		copiedApkFiles = append(copiedApkFiles, deployPth)
+		switch strings.ToLower(ext) {
+		case ".apk":
+			copiedApkFiles = append(copiedApkFiles, deployPth)
+		case ".aab":
+			copiedAabFiles = append(copiedAabFiles, deployPth)
+		default:
+			copiedApkFiles = append(copiedApkFiles, deployPth)
+		}
 	}
 
-	if lastCopiedApkFile != "" {
-		if err := exportEnvironmentWithEnvman("BITRISE_APK_PATH", lastCopiedApkFile); err != nil {
-			failf("Failed to export enviroment (BITRISE_APK_PATH), error: %s", err)
+	for appEnv, appFiles := range map[string][]string{
+		"BITRISE_APK_PATH": copiedApkFiles,
+		"BITRISE_AAB_PATH": copiedAabFiles} {
+		if len(appFiles) != 0 {
+			lastCopiedFile := appFiles[len(appFiles)-1]
+			if err := exportEnvironmentWithEnvman(appEnv, lastCopiedFile); err != nil {
+				failf("Failed to export enviroment (%s), error: %s", appEnv, err)
+			}
+			log.Donef("The app path is now available in the Environment Variable: $%s (value: %s)", appEnv, lastCopiedFile)
 		}
-		log.Donef("The apk path is now available in the Environment Variable: $BITRISE_APK_PATH (value: %s)", lastCopiedApkFile)
 	}
-	if len(copiedApkFiles) > 0 {
-		apkList := strings.Join(copiedApkFiles, "|")
-		if err := exportEnvironmentWithEnvman("BITRISE_APK_PATH_LIST", apkList); err != nil {
-			failf("Failed to export enviroment (BITRISE_APK_PATH_LIST), error: %s", err)
+	for appListEnv, appFiles := range map[string][]string{
```

Comment:
could not it be merged with the above for loop?

Reply:
No, appending _LIST to the env vars manuallt is hard to read.